### ISSUE TITLE
TastyInspector.inspectTastyFilesInJar doesn't  inspect jar

### ISSUE THREAD

**Issue by dos65** [lampepfl/dotty#11696]
## Compiler version

3.0.0-RC1

## Minimized code

I took the code from some files in `tests` directory.

The only difference is that test files from dotty repo extract jars somehow from classpath.
I tried to run it against local jars.

```Scala
import scala.quoted.*
import scala.tasty.inspector.*

@main def Test = {
  val inspector = new Inspector {
    def inspect(using Quotes)(tastys: List[Tasty[quotes.type]]): Unit = {
      println("Size:" + tastys.size.toString)
      for tasty <- tastys do
        tasty.ast.show(using quotes.reflect.Printer.TreeStructure)
    }
  }

  val jar = "$any_jar_with_tasty" // for example `cs fetch org.scala-lang:scala3-compiler_3.0.0-RC1:3.0.0-RC1 | grep scala3-compiler`
  TastyInspector.inspectTastyFilesInJar(jar)(inspector)
}
```

## Output

```scala
Size: 0
```

## Expectation

`scala3-compiler` has tasty files so there should be some `Tasty` passed into `inspect`

**Comment by jodersky** [comment]
I came across something similar. I think that `TastyInspector.inspectTastyFilesInJar` somehow only considers "top-level" entries in the jar. For example, it will find `foo.tasty`, but not `pkg/foo.tasty`. In other words, only things in the root package will be seen when reading from a jar.

I don't know if this is intentional, but the current behavior is certainly surprising

**Comment by nicolasstucki** [comment]
It should be reading tastys within packages.

**Comment by jodersky** [comment]
Should I create a new issue? I can confirm that under Scala 3.0.0, the tasty inspector only finds tasty files in jars if they are at the top level (i.e. their zip file entry does not contain a slash).

**Comment by nicolasstucki** [comment]
Yes, we should have another issue

**Comment by bishabosha** [comment]
is this fixed by #13254 @romanowski ?

**Comment by nicolasstucki** [comment]
Yes, this was fixed by #13254
